homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -133,12 +133,13 @@ def __init__(
 
     @callback
     def _update_attr(self, surepy_entity: SurepyEntity) -> None:
-        state = surepy_entity.raw_data()["status"]
-        self._attr_is_on = bool(state)
-        if state:
-            self._attr_extra_state_attributes = {
-                "device_rssi": f"{state['signal']['device_rssi']:.2f}",
-                "hub_rssi": f"{state['signal']['hub_rssi']:.2f}",
-            }
-        else:
-            self._attr_extra_state_attributes = {}
+        state = surepy_entity.raw_data().get("status", {})
+        online = bool(state.get("online", False))
+        self._attr_is_on = online
+        self._attr_extra_state_attributes = {}
+        if online:
+            value = state.get("signal", {}).get("device_rssi")
+            self._attr_extra_state_attributes["device_rssi"] = f"{value:.2f}" if value else "Unknown"
+            hub_rssi = state.get("signal",{}).get("hub_rssi")
```

Comment:
```suggestion
            hub_rssi = state.get("signal", {}).get("hub_rssi")
```
super_nitpick_mode: on ;-)